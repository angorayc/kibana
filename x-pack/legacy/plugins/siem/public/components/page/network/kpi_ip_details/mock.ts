/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { KpiIpDetailsData } from '../../../../graphql/types';

export const mockData: { KpiIpDetails: KpiIpDetailsData } = {
  KpiIpDetails: {
    connections: 246511947,
    hosts: 3,
    sourcePackets: 12927108,
    sourcePacketsHistogram: [
      {
        x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
        y: 1811625,
      },
      {
        x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
        y: 4041503,
      },
    ],
    sourceByte: 81325168682,
    sourceByteHistogram: [
      {
        x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
        y: 11592691523,
      },
      {
        x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
        y: 25511969914,
      },
    ],
    destinationPackets: 0,
    destinationPacketsHistogram: [
      {
        x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
        y: 0,
      },
      {
        x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
    destinationByte: 0,
    destinationByteHistogram: [
      {
        x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
        y: 0,
      },
      {
        x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
  },
};
