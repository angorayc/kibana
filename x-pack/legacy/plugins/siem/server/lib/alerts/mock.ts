/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { defaultIndexPattern } from '../../../default_index_pattern';

export const mockAlertsHistogramDataResponse = 'mockAlertsHistogramDataResponse';
export const mockAlertsHistogramQueryDsl = 'mockAlertsHistogramQueryDsl';
export const mockRequest = 'mockRequest';
export const mockOptions = {
  sourceConfiguration: { field: {} },
  timerange: {
    to: 9999,
    from: 1234,
  },
  defaultIndex: defaultIndexPattern,
  filterQuery: '',
};

// export const mockAlertsQueryDsl = {
//   mockAlertsQueryDsl: 'mockAlertsQueryDsl',
// };

// const mockTimelineAlertsInspectResponse = cloneDeep(mockAlertsResponse);
// delete mockTimelineAlertsInspectResponse.hits.hits[0]._source;
