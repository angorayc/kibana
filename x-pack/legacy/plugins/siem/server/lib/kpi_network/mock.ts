/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { RequestBasicOptions } from '../framework/types';

export const mockOptions: RequestBasicOptions = {
  defaultIndex: ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'],
  sourceConfiguration: {
    fields: {
      container: 'docker.container.name',
      host: 'beat.hostname',
      message: ['message', '@message'],
      pod: 'kubernetes.pod.name',
      tiebreaker: '_doc',
      timestamp: '@timestamp',
    },
  },
  timerange: { interval: '12h', to: 1549852006071, from: 1549765606071 },
  filterQuery: {},
};

export const mockMsearchOptions = {
  body: [],
};

export const mockRequest = {
  params: {},
  payload: {
    operationName: 'GetKpiNetworkQuery',
    variables: {
      sourceId: 'default',
      timerange: { interval: '12h', from: 1557445721842, to: 1557532121842 },
      filterQuery: '',
    },
    query:
      'fragment KpiNetworkChartFields on KpiNetworkHistogramData {\n  x\n  y\n  __typename\n}\n\nquery GetKpiNetworkQuery($sourceId: ID!, $timerange: TimerangeInput!, $filterQuery: String, $defaultIndex: [String!]!) {\n  source(id: $sourceId) {\n    id\n    KpiNetwork(timerange: $timerange, filterQuery: $filterQuery, defaultIndex: $defaultIndex) {\n      networkEvents\n      uniqueFlowId\n      uniqueSourcePrivateIps\n      uniqueSourcePrivateIpsHistogram {\n        ...KpiNetworkChartFields\n        __typename\n      }\n      uniqueDestinationPrivateIps\n      uniqueDestinationPrivateIpsHistogram {\n        ...KpiNetworkChartFields\n        __typename\n      }\n      dnsQueries\n      tlsHandshakes\n      __typename\n    }\n    __typename\n  }\n}\n',
  },
  query: {},
};

export const mockResponse = {
  responses: [
    {
      took: 384,
      timed_out: false,
      _shards: {
        total: 10,
        successful: 10,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 733106,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      status: 200,
    },
    {
      took: 64,
      timed_out: false,
      _shards: {
        total: 10,
        successful: 10,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 10942,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      status: 200,
    },
    {
      took: 224,
      timed_out: false,
      _shards: {
        total: 10,
        successful: 10,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 480755,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      aggregations: {
        source: {
          histogram: {
            buckets: [
              {
                key_as_string: '2019-05-09T23:00:00.000Z',
                key: 1557442800000,
                doc_count: 42109,
                count: {
                  value: 14,
                },
              },
              {
                key_as_string: '2019-05-10T11:00:00.000Z',
                key: 1557486000000,
                doc_count: 437160,
                count: {
                  value: 385,
                },
              },
              {
                key_as_string: '2019-05-10T23:00:00.000Z',
                key: 1557529200000,
                doc_count: 1486,
                count: {
                  value: 7,
                },
              },
            ],
            interval: '12h',
          },
          unique_private_ips: {
            value: 387,
          },
        },
        destination: {
          histogram: {
            buckets: [
              {
                key_as_string: '2019-05-09T23:00:00.000Z',
                key: 1557442800000,
                doc_count: 36253,
                count: {
                  value: 11,
                },
              },
              {
                key_as_string: '2019-05-10T11:00:00.000Z',
                key: 1557486000000,
                doc_count: 421719,
                count: {
                  value: 877,
                },
              },
              {
                key_as_string: '2019-05-10T23:00:00.000Z',
                key: 1557529200000,
                doc_count: 1311,
                count: {
                  value: 7,
                },
              },
            ],
            interval: '12h',
          },
          unique_private_ips: {
            value: 878,
          },
        },
      },
      status: 200,
    },
    {
      took: 384,
      timed_out: false,
      _shards: {
        total: 10,
        successful: 10,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 733106,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      aggregations: {
        unique_flow_id: {
          value: 195415,
        },
      },
      status: 200,
    },
    {
      took: 57,
      timed_out: false,
      _shards: {
        total: 10,
        successful: 10,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 54482,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      status: 200,
    },
  ],
};

export const mockResult = {
  dnsQueries: 10942,
  networkEvents: 733106,
  tlsHandshakes: 54482,
  uniqueDestinationPrivateIps: 878,
  uniqueDestinationPrivateIpsHistogram: [
    {
      x: new Date('2019-05-09T23:00:00.000Z').valueOf(),
      y: 11,
    },
    {
      x: new Date('2019-05-10T11:00:00.000Z').valueOf(),
      y: 877,
    },
    {
      x: new Date('2019-05-10T23:00:00.000Z').valueOf(),
      y: 7,
    },
  ],
  uniqueFlowId: 195415,
  uniqueSourcePrivateIps: 387,
  uniqueSourcePrivateIpsHistogram: [
    {
      x: new Date('2019-05-09T23:00:00.000Z').valueOf(),
      y: 14,
    },
    {
      x: new Date('2019-05-10T11:00:00.000Z').valueOf(),
      y: 385,
    },
    {
      x: new Date('2019-05-10T23:00:00.000Z').valueOf(),
      y: 7,
    },
  ],
};

export const mockKpiIpDetailsRequestOptions: RequestBasicOptions = {
  defaultIndex: ['auditbeat-*', 'filebeat-*', 'packetbeat-*', 'winlogbeat-*'],
  sourceConfiguration: {
    fields: {
      container: 'docker.container.name',
      message: ['message', '@message'],
      pod: 'kubernetes.pod.name',
      tiebreaker: '_doc',
      timestamp: '@timestamp',
      host: 'beat.hostname',
    },
  },
  timerange: { interval: '12h', to: 1549852006071, from: 1549765606071 },
  filterQuery: { term: { ip: '192.168.1.1' } },
};

export const mockKpiIpDetailsRequest = {
  params: {},
  payload: {
    operationName: 'GetKpiIpDetailsQuery',
    variables: {
      sourceId: 'default',
      timerange: { interval: '12h', from: 1557445721842, to: 1557532121842 },
      filterQuery: { term: { ip: '10.142.0.10' } },
    },
    query:
      'fragment KpiIpDetailsChartFields on KpiIpDetailsHistogramData {\n  x\n  y\n  __typename\n}\n\nquery GetKpiIpDetailsQuery($sourceId: ID!, $timerange: TimerangeInput!, $filterQuery: String, $defaultIndex: [String!]!, $ip: String!) {\n  source(id: $sourceId) {\n    id\n    KpiIpDetails(timerange: $timerange, filterQuery: $filterQuery, defaultIndex: $defaultIndex, ip: $ip) {\n      connections\n      hosts\n      sourcePackets\n      sourcePacketsHistogram {\n        ...KpiIpDetailsChartFields\n        __typename\n      }\n      sourceByte\n      sourceByteHistogram {\n        ...KpiIpDetailsChartFields\n        __typename\n      }\n      destinationPackets\n      destinationPacketsHistogram {\n        ...KpiIpDetailsChartFields\n        __typename\n      }\n      destinationByte\n      destinationByteHistogram {\n        ...KpiIpDetailsChartFields\n        __typename\n      }\n      __typename\n    }\n    __typename\n  }\n}\n',
  },
  query: {},
};

export const mockKpiIpDetailsResponse = {
  responses: [
    {
      took: 4295,
      timed_out: false,
      _shards: {
        total: 31,
        successful: 31,
        skipped: 0,
        failed: 0,
      },
      hits: {
        total: {
          value: 228246187,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      aggregations: {
        connections: {
          value: 13368,
        },
        destination: {
          doc_count: 6454,
          packetsHistogram: {
            buckets: [
              {
                key_as_string: '2019-06-14T00:00:00.000Z',
                key: 1560470400000,
                doc_count: 1775,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-15T00:00:00.000Z',
                key: 1560556800000,
                doc_count: 1763,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-16T00:00:00.000Z',
                key: 1560643200000,
                doc_count: 1796,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-17T00:00:00.000Z',
                key: 1560729600000,
                doc_count: 1111,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-18T00:00:00.000Z',
                key: 1560816000000,
                doc_count: 9,
                count: {
                  value: 0,
                },
              },
            ],
            interval: '1d',
          },
          bytes: {
            value: 0,
          },
          hosts: {
            value: 4,
          },
          bytesHistogram: {
            buckets: [
              {
                key_as_string: '2019-06-14T00:00:00.000Z',
                key: 1560470400000,
                doc_count: 1775,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-15T00:00:00.000Z',
                key: 1560556800000,
                doc_count: 1763,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-16T00:00:00.000Z',
                key: 1560643200000,
                doc_count: 1796,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-17T00:00:00.000Z',
                key: 1560729600000,
                doc_count: 1111,
                count: {
                  value: 0,
                },
              },
              {
                key_as_string: '2019-06-18T00:00:00.000Z',
                key: 1560816000000,
                doc_count: 9,
                count: {
                  value: 0,
                },
              },
            ],
            interval: '1d',
          },
          packets: {
            value: 0,
          },
        },
        source: {
          doc_count: 6914,
          packetsHistogram: {
            buckets: [
              {
                key_as_string: '2019-06-14T00:00:00.000Z',
                key: 1560470400000,
                doc_count: 1898,
                count: {
                  value: 5556185,
                },
              },
              {
                key_as_string: '2019-06-15T00:00:00.000Z',
                key: 1560556800000,
                doc_count: 1896,
                count: {
                  value: 2669758,
                },
              },
              {
                key_as_string: '2019-06-16T00:00:00.000Z',
                key: 1560643200000,
                doc_count: 1908,
                count: {
                  value: 2612273,
                },
              },
              {
                key_as_string: '2019-06-17T00:00:00.000Z',
                key: 1560729600000,
                doc_count: 1203,
                count: {
                  value: 1710466,
                },
              },
              {
                key_as_string: '2019-06-18T00:00:00.000Z',
                key: 1560816000000,
                doc_count: 9,
                count: {
                  value: 26540,
                },
              },
            ],
            interval: '1d',
          },
          bytes: {
            value: 80647226799,
          },
          hosts: {
            value: 4,
          },
          bytesHistogram: {
            buckets: [
              {
                key_as_string: '2019-06-14T00:00:00.000Z',
                key: 1560470400000,
                doc_count: 1898,
                count: {
                  value: 35934887439,
                },
              },
              {
                key_as_string: '2019-06-15T00:00:00.000Z',
                key: 1560556800000,
                doc_count: 1896,
                count: {
                  value: 17055922318,
                },
              },
              {
                key_as_string: '2019-06-16T00:00:00.000Z',
                key: 1560643200000,
                doc_count: 1908,
                count: {
                  value: 16677974647,
                },
              },
              {
                key_as_string: '2019-06-17T00:00:00.000Z',
                key: 1560729600000,
                doc_count: 1203,
                count: {
                  value: 10814051075,
                },
              },
              {
                key_as_string: '2019-06-18T00:00:00.000Z',
                key: 1560816000000,
                doc_count: 9,
                count: {
                  value: 164391320,
                },
              },
            ],
            interval: '1d',
          },
          packets: {
            value: 12575222,
          },
        },
      },
      status: 200,
    },
  ],
};

export const mockKpiIpDetailsResult = {
  connections: 13368,
  hosts: 4,
  sourcePackets: 12575222,
  sourcePacketsHistogram: [
    {
      x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
      y: 5556185,
    },
    {
      x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
      y: 2669758,
    },
    {
      x: new Date('2019-06-16T00:00:00.000Z').valueOf(),
      y: 2612273,
    },
    {
      x: new Date('2019-06-17T00:00:00.000Z').valueOf(),
      y: 1710466,
    },
    {
      x: new Date('2019-06-18T00:00:00.000Z').valueOf(),
      y: 26540,
    },
  ],
  sourceByte: 80647226799,
  sourceByteHistogram: [
    {
      x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
      y: 35934887439,
    },
    {
      x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
      y: 17055922318,
    },
    {
      x: new Date('2019-06-16T00:00:00.000Z').valueOf(),
      y: 16677974647,
    },
    {
      x: new Date('2019-06-17T00:00:00.000Z').valueOf(),
      y: 10814051075,
    },
    {
      x: new Date('2019-06-18T00:00:00.000Z').valueOf(),
      y: 164391320,
    },
  ],
  destinationPackets: 0,
  destinationPacketsHistogram: [
    {
      x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-16T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-17T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-18T00:00:00.000Z').valueOf(),
      y: 0,
    },
  ],
  destinationByte: 0,
  destinationByteHistogram: [
    {
      x: new Date('2019-06-14T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-15T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-16T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-17T00:00:00.000Z').valueOf(),
      y: 0,
    },
    {
      x: new Date('2019-06-18T00:00:00.000Z').valueOf(),
      y: 0,
    },
  ],
};
