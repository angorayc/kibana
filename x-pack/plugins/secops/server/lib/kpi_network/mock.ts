/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { RequestBasicOptions } from '../framework/types';

export const mockOptions: RequestBasicOptions = {
  sourceConfiguration: {
    logAlias: 'filebeat-*',
    auditbeatAlias: 'auditbeat-*',
    packetbeatAlias: 'packetbeat-*',
    fields: {
      container: 'docker.container.name',
      host: 'beat.hostname',
      message: ['message', '@message'],
      pod: 'kubernetes.pod.name',
      tiebreaker: '_doc',
      timestamp: '@timestamp',
    },
  },
  timerange: { interval: '12h', to: 1549852006071, from: 1549765606071 },
  filterQuery: {},
};

export const mockRequest = {
  params: {},
  payload: {
    operationName: 'GetKpiNetworkQuery',
    variables: {
      sourceId: 'default',
      timerange: { interval: '12h', from: 1549765830772, to: 1549852230772 },
      filterQuery: '',
    },
    query:
      'query GetKpiNetworkQuery($sourceId: ID!, $timerange: TimerangeInput!, $filterQuery: String) {\n  source(id: $sourceId) {\n    id\n     KpiNetwork(timerange: $timerange, filterQuery: $filterQuery) {\n       networkEvents\n       uniqueFlowId\n       activeAgents\n     }\n   }\n }',
  },
  query: {},
};

export const mockResponse = {
  took: 89,
  timed_out: false,
  _shards: { total: 18, successful: 18, skipped: 0, failed: 0 },
  hits: { total: { value: 950867, relation: 'eq' }, max_score: null, hits: [] },
  aggregations: { unique_flow_id: { value: 50243 }, active_agents: { value: 15 } },
};

export const mockResult = {
  networkEvents: 950867,
  uniqueFlowId: 50243,
  activeAgents: 15,
};
