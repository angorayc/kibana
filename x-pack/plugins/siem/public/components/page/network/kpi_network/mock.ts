/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { KpiNetworkData } from '../../../../graphql/types';

export const mockData: { KpiNetwork: KpiNetworkData } = {
  KpiNetwork: {
    networkEvents: 16,
    uniqueFlowId: 10277307,
    uniqueSourcePrivateIps: 383,
    uniqueSourcePrivateIpsHistogram: [
      {
        x: new Date('2019-02-09T16:00:00.000Z').valueOf(),
        y: 8,
      },
      {
        x: new Date('2019-02-09T19:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
    uniqueDestinationPrivateIps: 18,
    uniqueDestinationPrivateIpsHistogram: [
      {
        x: new Date('2019-02-09T16:00:00.000Z').valueOf(),
        y: 8,
      },
      {
        x: new Date('2019-02-09T19:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
    dnsQueries: 278,
    tlsHandshakes: 10000,
  },
};
