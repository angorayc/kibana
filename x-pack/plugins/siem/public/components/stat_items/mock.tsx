/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { StatItems, KpiValue } from '.';
import { KpiNetworkData } from '../../graphql/types';

export const mockRenderer = (value: KpiValue) => value;

export const mockMappings: Readonly<Array<StatItems<KpiValue>>> = [
  {
    key: 'UniqueIps',
    fields: [
      {
        key: 'uniqueSourcePrivateIps',
        value: null,
        name: 'Src.',
        description: 'Source',
        color: '#DB1374',
        icon: 'visMapCoordinate',
        render: mockRenderer,
      },
      {
        key: 'uniqueDestinationPrivateIps',
        value: null,
        name: 'Dist.',
        description: 'Destination',
        color: '#490092',
        icon: 'visMapCoordinate',
        render: mockRenderer,
      },
    ],
    description: 'Unique Private IPs',
    enableAreaChart: true,
    enableBarChart: true,
    grow: 2,
  },
];

export const mockData: { KpiNetwork: KpiNetworkData } = {
  KpiNetwork: {
    networkEvents: 16,
    uniqueFlowId: 10277307,
    uniqueSourcePrivateIps: 383,
    uniqueSourcePrivateIpsHistogram: [
      {
        x: new Date('2019-02-09T16:00:00.000Z').valueOf(),
        y: 8,
      },
      {
        x: new Date('2019-02-09T19:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
    uniqueDestinationPrivateIps: 18,
    uniqueDestinationPrivateIpsHistogram: [
      {
        x: new Date('2019-02-09T16:00:00.000Z').valueOf(),
        y: 8,
      },
      {
        x: new Date('2019-02-09T19:00:00.000Z').valueOf(),
        y: 0,
      },
    ],
    dnsQueries: 278,
    tlsHandshakes: 10000,
  },
};

export const mockEnableChartsData = {
  fields: [
    {
      key: 'uniqueSourcePrivateIps',
      value: 383,
      name: 'Src.',
      description: 'Source',
      color: '#DB1374',
      icon: 'visMapCoordinate',
      render: mockRenderer,
    },
    {
      key: 'uniqueDestinationPrivateIps',
      value: 18,
      name: 'Dist.',
      description: 'Destination',
      color: '#490092',
      icon: 'visMapCoordinate',
      render: mockRenderer,
    },
  ],
  description: 'Unique Private IPs',
  enableAreaChart: true,
  enableBarChart: true,
  grow: 2,
  areaChart: [
    {
      key: 'uniqueSourcePrivateIpsHistogram',
      value: [
        { x: new Date('2019-02-09T16:00:00.000Z').valueOf(), y: 8 },
        {
          x: new Date('2019-02-09T19:00:00.000Z').valueOf(),
          y: 0,
        },
      ],
      name: 'Src.',
      description: 'Source',
      color: '#DB1374',
      icon: 'visMapCoordinate',
    },
    {
      key: 'uniqueDestinationPrivateIpsHistogram',
      value: [
        { x: new Date('2019-02-09T16:00:00.000Z').valueOf(), y: 8 },
        { x: new Date('2019-02-09T19:00:00.000Z').valueOf(), y: 0 },
      ],
      name: 'Dist.',
      description: 'Destination',
      color: '#490092',
      icon: 'visMapCoordinate',
    },
  ],
  barChart: [
    {
      key: 'uniqueSourcePrivateIps',
      color: '#DB1374',
      value: [{ x: 'Src.', y: 383, g: 'uniqueSourcePrivateIps' }],
    },
    {
      key: 'uniqueDestinationPrivateIps',
      color: '#490092',
      value: [{ x: 'Dist.', y: 18, g: 'uniqueDestinationPrivateIps' }],
    },
  ],
};

const mockMappingItems: Readonly<StatItems<KpiValue>> = {
  key: 'UniqueIps',
  fields: [
    {
      key: 'uniqueSourcePrivateIps',
      value: null,
      name: 'Src.',
      description: 'Source',
      color: '#DB1374',
      icon: 'visMapCoordinate',
      render: mockRenderer,
    },
    {
      key: 'uniqueDestinationPrivateIps',
      value: null,
      name: 'Dist.',
      description: 'Destination',
      color: '#490092',
      icon: 'visMapCoordinate',
      render: mockRenderer,
    },
  ],
  description: 'Unique Private IPs',
  enableAreaChart: true,
  enableBarChart: true,
  grow: 2,
};

export const mockNoChartMappings: Readonly<Array<StatItems<KpiValue>>> = [
  {
    ...mockMappingItems,
    enableAreaChart: false,
    enableBarChart: false,
  },
];
