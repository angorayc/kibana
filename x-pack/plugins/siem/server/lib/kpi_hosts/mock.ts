/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { RequestBasicOptions } from '../framework/types';

export const mockOptions: RequestBasicOptions = {
  sourceConfiguration: {
    logAlias: 'filebeat-*',
    auditbeatAlias: 'auditbeat-*',
    packetbeatAlias: 'packetbeat-*',
    winlogbeatAlias: 'winlogbeat-*',
    fields: {
      container: 'docker.container.name',
      host: 'beat.hostname',
      message: ['message', '@message'],
      pod: 'kubernetes.pod.name',
      tiebreaker: '_doc',
      timestamp: '@timestamp',
    },
  },
  timerange: { interval: '12h', to: 1549852006071, from: 1549765606071 },
  filterQuery: {},
};

export const mockRequest = {
  params: {},
  payload: {
    operationName: 'GetKpiHostsQuery',
    variables: {
      sourceId: 'default',
      timerange: { interval: '12h', from: 1549765830772, to: 1549852230772 },
      filterQuery: '',
    },
    query:
      'query GetKpiHostsQuery($sourceId: ID!, $timerange: TimerangeInput!, $filterQuery: String) {\n  source(id: $sourceId) {\n    id\n    KpiHosts(timerange: $timerange, filterQuery: $filterQuery) {\n      hosts\n      agents\n      authentication {\n        success\n        failure\n        __typename\n      }\n      uniqueSourceIps\n      uniqueDestinationIps\n      __typename\n    }\n    __typename\n  }\n}\n',
  },
  query: {},
};

export const mockResponse = {
  took: 577,
  responses: [
    {
      took: 2603,
      timed_out: false,
      _shards: {
        total: 67,
        successful: 67,
        skipped: 60,
        failed: 0,
      },
      hits: {
        total: {
          value: 9665113,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      aggregations: {
        unique_source_ips: {
          value: 10503,
        },
        hosts: {
          value: 711,
        },
        unique_destination_ips: {
          value: 2380,
        },
        agents: {
          value: 23,
        },
      },
      status: 200,
    },
    {
      took: 3884,
      timed_out: false,
      _shards: {
        total: 67,
        successful: 67,
        skipped: 60,
        failed: 0,
      },
      hits: {
        total: {
          value: 661651,
          relation: 'eq',
        },
        max_score: null,
        hits: [],
      },
      aggregations: {
        authentication_success: {
          doc_count: 2,
        },
        authentication_failure: {
          doc_count: 661649,
        },
      },
      status: 200,
    },
  ],
};

export const mockResult = {
  hosts: 711,
  agents: 23,
  authentication: {
    success: 2,
    failure: 661649,
  },
  uniqueSourceIps: 10503,
  uniqueDestinationIps: 2380,
};

export const mockGeneralQuery = [
  {
    index: ['filebeat-*', 'auditbeat-*', 'packetbeat-*', 'winlogbeat-*'],
    allowNoIndices: true,
    ignoreUnavailable: true,
  },
  {
    aggregations: {
      hosts: { cardinality: { field: 'host.name' } },
      agents: { cardinality: { field: 'agent.id' } },
      unique_source_ips: { cardinality: { field: 'source.ip' } },
      unique_destination_ips: { cardinality: { field: 'destination.ip' } },
    },
    query: {
      bool: { filter: [{ range: { '@timestamp': { gte: 1549765606071, lte: 1549852006071 } } }] },
    },
    size: 0,
    track_total_hits: false,
  },
];

export const mockAuthQuery = [
  {
    index: ['filebeat-*', 'auditbeat-*', 'packetbeat-*', 'winlogbeat-*'],
    allowNoIndices: true,
    ignoreUnavailable: true,
  },
  {
    aggs: {
      authentication_success: {
        filter: { term: { 'event.type': 'authentication_success' } },
        aggs: { attempts_over_time: { auto_date_histogram: { field: '@timestamp', buckets: 10 } } },
      },
      authentication_failure: {
        filter: { term: { 'event.type': 'authentication_failure' } },
        aggs: { attempts_over_time: { auto_date_histogram: { field: '@timestamp', buckets: 10 } } },
      },
    },
    query: {
      bool: {
        filter: [
          {
            bool: {
              should: [
                { match: { 'event.type': 'authentication_success' } },
                { match: { 'event.type': 'authentication_failure' } },
              ],
              minimum_should_match: 1,
            },
          },
          { range: { '@timestamp': { gte: 1549765606071, lte: 1549852006071 } } },
        ],
      },
    },
    size: 0,
    track_total_hits: true,
  },
];

export const mockMsearchOptions = {
  body: [...mockGeneralQuery, ...mockAuthQuery],
};
