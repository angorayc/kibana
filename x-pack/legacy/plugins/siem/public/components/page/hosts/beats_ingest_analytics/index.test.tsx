/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';

import { mockData } from './mock';
import { BeatsIngestAnalytics } from '.';

describe('beats ingest analytics', () => {
  describe('render', () => {
    test('it should render spinner if it is loading', () => {
      const wrapper: ShallowWrapper = shallow(
        <BeatsIngestAnalytics data={mockData} loading={true} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('it should render BeatsIngestAnalytics', () => {
      const wrapper: ShallowWrapper = shallow(
        <BeatsIngestAnalytics data={mockData} loading={false} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
