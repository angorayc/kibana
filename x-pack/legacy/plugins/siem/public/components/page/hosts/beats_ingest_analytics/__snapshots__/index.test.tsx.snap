// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`beats ingest analytics render it should render BeatsIngestAnalytics 1`] = `
<EuiPopover
  anchorPosition="upCenter"
  button={
    <EuiButtonEmpty
      color="primary"
      iconSide="right"
      iconType="arrowDown"
      onClick={[Function]}
      type="button"
    >
      <EuiText
        size="s"
      >
        Beats Ingest Analytics
      </EuiText>
    </EuiButtonEmpty>
  }
  closePopover={[Function]}
  hasArrow={true}
  id="beats-ingest-analytics"
  isOpen={false}
  ownFocus={true}
  panelPaddingSize="m"
>
  <styled.div
    key="beats-ingest-analytics-auditbeat"
  >
    <EuiText
      size="s"
    >
      <EuiFlexGroup
        gutterSize="s"
        justifyContent="spaceBetween"
      >
        <EuiFlexItem
          grow={1}
        >
          Auditbeat Events
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          10
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiText>
  </styled.div>
  <styled.div
    key="beats-ingest-analytics-filebeat"
  >
    <EuiText
      size="s"
    >
      <EuiFlexGroup
        gutterSize="s"
        justifyContent="spaceBetween"
      >
        <EuiFlexItem
          grow={1}
        >
          Filebeat Events
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          30
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiText>
  </styled.div>
  <styled.div
    key="beats-ingest-analytics-winlogbeat"
  >
    <EuiText
      size="s"
    >
      <EuiFlexGroup
        gutterSize="s"
        justifyContent="spaceBetween"
      >
        <EuiFlexItem
          grow={1}
        >
          Winlogbeat Events
        </EuiFlexItem>
        <EuiFlexItem
          grow={false}
        >
          20
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiText>
  </styled.div>
</EuiPopover>
`;

exports[`beats ingest analytics render it should render spinner if it is loading 1`] = `""`;
